src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,55 @@
+package duke.command;
+
+import duke.logic.DukeException;
+import duke.logic.Storage;
+import duke.logic.TaskList;
+import duke.logic.Ui;
+import duke.task.Task;
+
+/**
+ * Represents a command that adds a task.
+ *
+ * @author Peter
+ */
+public class AddCommand extends Command {
+    /**
+     * Task that is to be added.
+     */
+    private final Task task;
+
+    /**
+     * Constructor for a task add command.
+     *
+     * @param task Task that is to be added.
+     */
+    public AddCommand(Task task) {
+        super();
+        this.task = task;
+    }
+
+    /**
+     * Adds the associated task to a given list of tasks, displays a response message, and writes
+     * to a local file associated with the list of tasks.
+     *
+     * @param taskList List of tasks that is to be appended.
+     * @param ui       UI responsible for displaying the response message.
+     * @param storage  Storage responsible for writing to local file.
+     * @return <code>true</code> upon successful execution.
+     * @throws DukeException If write to file is unsuccessful.
+     */
+    @Override
+    public String execute(TaskList taskList, Ui ui, Storage storage) throws DukeException {
+        taskList.add(this.task);
+        storage.writeToFile(taskList);
+        String output = "TASK ADDED:\n"
+                + task + "\n"
+                + taskList.size() + " TASK(S) NOW.";
```

Comment:
`output` could be defined as a private field of the AddCommnad class instead of being defined here. This will allow for a Single Layer of Abstraction (SLAP) in this method and makes it easier to find where to change the output message if needed in the future. What do you think?